File: cpp/32.cpp
#include <iostream>
#include <algorithm>
#include <set>

using namespace std;


int main() {
    set<int> pandigitals {};
    string p {};
    int sum {};

    for (int i = 1; i < 9999; i++)
    {
        for (int j = 0; j < 9999; j++)
        {
            p = to_string(i * j) + to_string(i) + to_string(j);
            if (p.length() > 9)
                break;

            sort(p.begin(), p.end());
            if (p == "123456789")
                pandigitals.insert(i * j);
        }
    }
    
    for (int i: pandigitals)
        sum += i;

    cout << sum << endl;
    return 0;
}

File: python/32.py

pandigitals = []
for i in range(1, 9999):
    for j in range(1, 9999):
        p = str(i * j)
        t = p + str(i) + str(j)
        if len(t) > 9:
            break
        if len(t) == 9 and sorted(t) == ['1', '2', '3', '4', '5', '6', '7', '8', '9']:
            pandigitals.append(int(p))

print(sum(set(pandigitals)))
